extra/mariabackup/encryption_plugin.cc

```diff
@@ -161,6 +168,14 @@ void encryption_plugin_backup_init(MYSQL *mysql)
     oss << "innodb_encrypt_log=" << row[0] << std::endl;
 
     mysql_free_result(result);
+
+    /* Check whether to encrypt tables */
+    result = xb_mysql_query(mysql, "select @@innodb_encrypt_tables", true, true);
+    row = mysql_fetch_row(result);
+    srv_encrypt_tables = (row != 0 && row[0][0] == '1');
+    oss << "innodb_encrypt_tables=" << row[0] << std::endl;
```

Comment:
This seems to be interpreting any response starting with `1` as true. When can `mysql_fetch_row` return `nullptr`, or when can it return a pointer to `nullptr`? The check `row != 0` looks redundant, because in the following line we are dereferencing `row` anyway. And we fail to check if `row[0]` is a null pointer.

With some error handling (I am not sure if it is correct or redundant), it could be clearer to write
```c++
if (!row);
else if (const char *r= row[0])
{
  srv_encrypt_tables = !strcmp(r, "1");
  oss << "innodb_encrypt_tables=" << r << std::endl
}
```